Mayhem token is optional for mayhem.forallsecure.com (#10)
ForAllSecure/mcode-action: README.md

To use the Mayhem for Code GitHub Action, perform the following steps:

> 1.  Navigate to [mayhem.forallsecure.com](https://mayhem.forallsecure.com/) to register an account.

Reviewer: ```suggestion
1. Navigate to [mayhem.forallsecure.com](https://mayhem.forallsecure.com/) to register an account.
```

      - name: Start analysis
        uses: ForAllSecure/mcode-action@v1
        with:
          args: --image ${{ steps.meta.outputs.tags }}
          sarif-output: results.sarif

      - name: Upload SARIF file(s)
        uses: github/codeql-action/upload-sarif@v1
        with:
          sarif_file: results.sarif
```

The mCode Action accepts the following inputs:

| Required | Input Name | Type | Description | Default
| --- | --- | --- | --- | ---
|   | `mayhem-url` | string | Path to a custom Mayhem for Code instance. | https://mayhem.forallsecure.com |
|   | `mayhem-token` | string | Mayhem for Code account token. Only required if overriding `mayhem-url` |

> |   | `args` | string | Additional arguments such as specifying the `--corpus` directory path |

Reviewer: ```suggestion
|   | `args` | string | Additional [arguments](https://mayhem.forallsecure.com/docs/mayhem-cli/getting-started/mayhem-cli-commands/#run) such as specifying the `--tests` directory path |
```

| Required | Input Name | Type | Description | Default
| --- | --- | --- | --- | ---
|   | `mayhem-url` | string | Path to a custom Mayhem for Code instance. | https://mayhem.forallsecure.com |
|   | `mayhem-token` | string | Mayhem for Code account token. Only required if overriding `mayhem-url` |
|   | `args` | string | Additional CLI override [arguments](https://mayhem.forallsecure.com/docs/mayhem-cli/getting-started/mayhem-cli-commands/#run) such as specifying the `--tests` directory path |
|   | `sarif-output` | string | Path for generating a SARIF report output file |

## Reports and GitHub Code Scanning

Mayhem for Code generates [SARIF reports](https://sarifweb.azurewebsites.net/#:~:text=The%20Static%20Analysis%20Results%20Interchange,approved%20as%20an%20OASIS%20standard.) for your application security testing results.

SARIF reports are generated using the `sarif-output` parameter, which specifies an output file path.

> To upload the SARIF report to GitHub, use the `github/codeql-action/upload-sarif@v1` action with the `sarif_file` parameter to specify the location of an input SARIF file to upload to GitHub.

Author: ```suggestion
To upload the SARIF report to GitHub, use the `github/codeql-action/upload-sarif@v1` action with the `sarif_file` parameter to specify the location of a path containing SARIF results to upload to GitHub.
```
